### PR TITLE
Fix VP9.2 config verification

### LINKED ISSUE #534: BSW/BDW test_va_api GetCreateConfig/VAAPIGetCreateConfig tests failing for VP9 Profile 2
Hi friends, the following test cases have been failing on BSW and BDW Chromebooks. These two chipsets are built with the hybrid_codec enabled.

```
localhost ~ # test_va_api
<snip>
[----------] Global test environment tear-down
[==========] 17415 tests from 14 test cases ran. (67294 ms total)
[  PASSED  ] 17395 tests.
[  FAILED  ] 20 tests, listed below:
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigWithAttributes/210, where GetParam() = (21:VAProfileVP9Profile2, 1:VAEntrypointVLD)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigWithAttributes/211, where GetParam() = (21:VAProfileVP9Profile2, 2:VAEntrypointIZZ)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigWithAttributes/212, where GetParam() = (21:VAProfileVP9Profile2, 3:VAEntrypointIDCT)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigWithAttributes/213, where GetParam() = (21:VAProfileVP9Profile2, 4:VAEntrypointMoComp)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigWithAttributes/214, where GetParam() = (21:VAProfileVP9Profile2, 5:VAEntrypointDeblocking)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigWithAttributes/215, where GetParam() = (21:VAProfileVP9Profile2, 6:VAEntrypointEncSlice)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigWithAttributes/216, where GetParam() = (21:VAProfileVP9Profile2, 7:VAEntrypointEncPicture)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigWithAttributes/217, where GetParam() = (21:VAProfileVP9Profile2, 8:VAEntrypointEncSliceLP)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigWithAttributes/218, where GetParam() = (21:VAProfileVP9Profile2, 10:VAEntrypointVideoProc)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigWithAttributes/219, where GetParam() = (21:VAProfileVP9Profile2, 11:VAEntrypointFEI)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigNoAttributes/210, where GetParam() = (21:VAProfileVP9Profile2, 1:VAEntrypointVLD)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigNoAttributes/211, where GetParam() = (21:VAProfileVP9Profile2, 2:VAEntrypointIZZ)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigNoAttributes/212, where GetParam() = (21:VAProfileVP9Profile2, 3:VAEntrypointIDCT)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigNoAttributes/213, where GetParam() = (21:VAProfileVP9Profile2, 4:VAEntrypointMoComp)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigNoAttributes/214, where GetParam() = (21:VAProfileVP9Profile2, 5:VAEntrypointDeblocking)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigNoAttributes/215, where GetParam() = (21:VAProfileVP9Profile2, 6:VAEntrypointEncSlice)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigNoAttributes/216, where GetParam() = (21:VAProfileVP9Profile2, 7:VAEntrypointEncPicture)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigNoAttributes/217, where GetParam() = (21:VAProfileVP9Profile2, 8:VAEntrypointEncSliceLP)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigNoAttributes/218, where GetParam() = (21:VAProfileVP9Profile2, 10:VAEntrypointVideoProc)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigNoAttributes/219, where GetParam() = (21:VAProfileVP9Profile2, 11:VAEntrypointFEI)

20 FAILED TESTS
```

### PR DESCRIPTION
**Pull request by edwarddavidbaker**
This commit addresses the following test_va_api failures collected on
a Braswell with the hybrid codec enabled. Fixes #534.

```
localhost ~ # test_va_api
<snip>
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigWithAttributes/210, where GetParam() = (21:VAProfileVP9Profile2, 1:VAEntrypointVLD)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigWithAttributes/211, where GetParam() = (21:VAProfileVP9Profile2, 2:VAEntrypointIZZ)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigWithAttributes/212, where GetParam() = (21:VAProfileVP9Profile2, 3:VAEntrypointIDCT)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigWithAttributes/213, where GetParam() = (21:VAProfileVP9Profile2, 4:VAEntrypointMoComp)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigWithAttributes/214, where GetParam() = (21:VAProfileVP9Profile2, 5:VAEntrypointDeblocking)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigWithAttributes/215, where GetParam() = (21:VAProfileVP9Profile2, 6:VAEntrypointEncSlice)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigWithAttributes/216, where GetParam() = (21:VAProfileVP9Profile2, 7:VAEntrypointEncPicture)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigWithAttributes/217, where GetParam() = (21:VAProfileVP9Profile2, 8:VAEntrypointEncSliceLP)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigWithAttributes/218, where GetParam() = (21:VAProfileVP9Profile2, 10:VAEntrypointVideoProc)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigWithAttributes/219, where GetParam() = (21:VAProfileVP9Profile2, 11:VAEntrypointFEI)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigNoAttributes/210, where GetParam() = (21:VAProfileVP9Profile2, 1:VAEntrypointVLD)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigNoAttributes/211, where GetParam() = (21:VAProfileVP9Profile2, 2:VAEntrypointIZZ)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigNoAttributes/212, where GetParam() = (21:VAProfileVP9Profile2, 3:VAEntrypointIDCT)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigNoAttributes/213, where GetParam() = (21:VAProfileVP9Profile2, 4:VAEntrypointMoComp)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigNoAttributes/214, where GetParam() = (21:VAProfileVP9Profile2, 5:VAEntrypointDeblocking)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigNoAttributes/215, where GetParam() = (21:VAProfileVP9Profile2, 6:VAEntrypointEncSlice)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigNoAttributes/216, where GetParam() = (21:VAProfileVP9Profile2, 7:VAEntrypointEncPicture)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigNoAttributes/217, where GetParam() = (21:VAProfileVP9Profile2, 8:VAEntrypointEncSliceLP)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigNoAttributes/218, where GetParam() = (21:VAProfileVP9Profile2, 10:VAEntrypointVideoProc)
[  FAILED  ] GetCreateConfig/VAAPIGetCreateConfig.CreateConfigNoAttributes/219, where GetParam() = (21:VAProfileVP9Profile2, 11:VAEntrypointFEI)
```

Signed-off-by: Ed Baker <edward.baker@intel.com>

### CODE REVIEW

**Comment by uartie** [comment]
> This patch is one possible solution. Also, I'm curious if CI/CD ever caught it

We don't test the "hybrid" driver in our internal CI since it's meant for fairly older hardware.

**Comment by edwarddavidbaker** [comment]
Hi @xhaihao, when you get a minute could you take a look? Thanks for the help.

File: src/i965_drv_video.c [hunk 1]
```diff
@@ -892,7 +892,9 @@ i965_validate_config(VADriverContextP ctx, VAProfile profile,
         } else if (!HAS_VP9_DECODING_PROFILE(i965, profile) &&
                    !HAS_VP9_ENCODING(i965) &&
                    !HAS_LP_VP9_ENCODING(i965) &&
-                   !i965->wrapper_pdrvctx) {
+                   (!i965->wrapper_pdrvctx ||
+                    /* Hybrid driver does not support VP9.2 */
+                    i965->wrapper_pdrvctx && profile == VAProfileVP9Profile2)) {
```
Comment (xhaihao) [inline src/i965_drv_video.c:897]: How about to call  vaQueryConfigEntrypoints to get the entrypoint list from the hybrid driver first, then check whether the specified entrypoint is included in the returned list  ? 

               CALL_VTABLE(pdrvctx, va_status,                                                                                                                                                
                            vaQueryConfigEntrypoints(pdrvctx, VAProfileVP9Profile2,
                                                     entrypoint_list,
                                                     num_entrypoints));

Comment (edwarddavidbaker) [reply src/i965_drv_video.c:897]: Thanks for the feedback. I'll give it a try.

I initially went with a simplistic approach to maintain consistency with an earlier if else block at https://github.com/intel/intel-vaapi-driver/blob/master/src/i965_drv_video.c#L888-L891 . If the wrapper context is set, it always returns success for VP9 profile 0 decode.

Comment (xhaihao) [reply src/i965_drv_video.c:897]: If you are willing, you may provide a patch to fix it for VP9 profile 0 too, thanks!

Comment (edwarddavidbaker) [reply src/i965_drv_video.c:897]: Hi, Sorry about the delays. I have an updated patch but still need to test it on a few more devices. Thanks.

**Comment by edwarddavidbaker** [comment]
I've verified test_va_api passes with the updated patch (9002fef) on a GLK N5000, a BSW N3150, and a KBL m3-7Y30. Please take a look when you get a minute. Thanks.

**Pull request merged**